ocaml/xapi/xapi_pool.ml

```
@@ -433,6 +440,7 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
   assert_management_vlan_are_same ();
   assert_external_auth_matches ();
   assert_restrictions_match ();
+  assert_pool_size_unrestricted ();
```

Let's move this up in the list, because it's quick, so that we can fail early.